crates/unified-bridge/src/network_id.rs

```diff
@@ -0,0 +1,55 @@
+use std::fmt::Display;
+
+use serde::{Deserialize, Serialize};
+
+/// A network ID.
+///
+/// 0 refers to ethereum, and rollups are numbered from 1 to `u32::MAX`.
+#[derive(Debug, Clone, Copy, PartialEq, Eq, PartialOrd, Ord, Serialize, Deserialize, Hash)]
+#[cfg_attr(feature = "testutils", derive(arbitrary::Arbitrary))]
+pub struct NetworkId(u32);
+
+impl Display for NetworkId {
+    #[inline]
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        self.0.fmt(f)
+    }
+}
+
+impl NetworkId {
+    pub const BITS: usize = u32::BITS as usize;
```

Comment:
I think it would make sense to introduce a constant (or arity 0 `const fn` for what I care) to stand for mainnet:
```suggestion
impl NetworkId {
    pub const BITS: usize = u32::BITS as usize;
    
    pub const MAINNET: Self = Self::new(0);
```

(btw I'm following the rollup index terminology here but it seems weird to use the word mainnet to refer to the L1)

Reply:
ah then we probably want to remove the already existing `L1_NETWORK_ID` defined here: https://github.com/agglayer/interop/blob/469b9b24cde2b8b40bef490f62dfcda8a8c08d76/crates/unified-bridge/src/token_info.rs#L7

Reply:
Done 😄